src/UCTNode.cpp

```diff
@@ -251,7 +251,8 @@ UCTNode* UCTNode::uct_select_child(int color, bool is_root) {
         }
     }
 
-    const auto numerator = std::sqrt(double(parentvisits));
+    const auto numerator = std::sqrt(double(parentvisits) *
+            std::log(cfg_logpuct * double(parentvisits) + cfg_logconst));
```

Comment:
Since `cfg_logpuct` and `cfg_logconst` are only used here, why not have them be `double` in the first place? Then you won't need any cast at all.